DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,283 @@
+
+Fork both of the following repositories to your own GitHub account:
+
+- [OpenSearch Dashboards](https://github.com/opensearch-project/OpenSearch-Dashboards)
+- [Query Insights Dashboards Plugin](https://github.com/opensearch-project/query-insights-dashboards)
+
+
+### Step 2: Clone OpenSearch Dashboards
+
+Clone your fork of **OpenSearch Dashboards** to your local machine:
+
+```bash
+git clone https://github.com/<your-username>/OpenSearch-Dashboards.git
+cd OpenSearch-Dashboards
+```
+
+
+### Step 3: Clone the Plugin into the plugins/ Directory
+Inside the OpenSearch-Dashboards directory:
+
+```
+cd plugins
+git clone https://github.com/<your-username>/query-insights-dashboards.git
+cd ..
+```
+
+### File Structure
+
+The file structure should be this
+
+```
+OpenSearch-Dashboards/
+├── plugins/
+    └── query-insights-dashboards/
+```
+
+
+
+##  Bootstrap OpenSearch Dashboards with the Query Insights Plugin
+
+After cloning both the OpenSearch Dashboards repository and placing the Query Insights plugin inside the `plugins/` folder, you need to bootstrap the environment. This installs all dependencies and registers the plugin with OpenSearch Dashboards.
+
+---
+
+### Step 1: Install Dependencies and Build
+
+From the root of your `OpenSearch-Dashboards\plugins` directory, run:
+
+```bash
+cd..
+yarn osd bootstrap
+```
+
+If you've previously bootstrapped the project and need to start fresh, first run:
+
+```bash
+$ yarn osd clean
+```
+
+##  Fork and Run Query Insights Plugin (Backend)
+
+The Query Insights Dashboards Plugin works in tandem with the backend Query Insights plugin. You must have the backend plugin installed and running within an OpenSearch cluster.
+
+
+
+### Step 1: Fork and Clone Query Insights Backend Plugin
+
+1. Fork the repository:  
+   [https://github.com/opensearch-project/query-insights](https://github.com/opensearch-project/query-insights)
+
+2. Clone your fork locally:
+
+```bash
+git clone https://github.com/<your-username>/query-insights.git
+cd query-insights
+```
+
+### Step 2: Run OpenSearch with the Plugin
+Use the included Gradle task to start a local OpenSearch cluster with the plugin auto-installed:
+
+```
+bash
+./gradlew run
+```
+
+ By default, the cluster runs at: http://localhost:9200
+
+Step 3: Verify Plugin Installation
+Check that the plugin is installed:
+
+```
+curl http://localhost:9200/_cat/plugins?v
+```
+
+You should see query-insights in the list.
+
+
+
+
+### Run OpenSearch Dashboards
+
+_**Warning:** Starting the OpenSearch Dashboards instance before the OpenSearch server is fully initialized can cause Dashboards to misbehave. Ensure that the OpenSearch server instance is up and running first. You can validate by running `curl localhost:9200` in another console tab or window (see [OpenSearch developer guide](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#run-opensearch))._
+
+Start the OpenSearch Dashboards development server:
+
+```bash
+$ yarn start
```

Comment:
Inconsistent usage of backticks in some places (like $ yarn start) - some use $ prefix, some don’t. Standardize all commands for consistency.